### PR TITLE
labels on X axis are now shown in full amount, also does not overlap

### PR DESCRIPTION
**Pull request by snapp1**

### CODE REVIEW

**Comment by snapp1** [comment]
@microsoft-github-policy-service agree [company="{Microsoft}"]

**Comment by snapp1** [comment]
@microsoft-github-policy-service agree

File: src/behavior.ts [hunk 1]
```diff
@@ -25,45 +25,63 @@
...
-    clearCatcher: Selection<d3.BaseType, any, any, any>;
-    interactivityService: IInteractivityService;
+export interface BehaviorOptions extends interactivityBaseService.IBehaviorOptions<StreamGraphSeries>{
+    selection: Selection<BaseType, StreamGraphSeries, any, any>;
+    clearCatcher: Selection<BaseType, any, any, any>;
+    interactivityService: IInteractivityService<any>;
     series: StreamGraphSeries[];
 }
 
 const getEvent = () => require("d3-selection").event;
```
Comment (MulyukovAidar) [inline src/behavior.ts:45]: is this used?

File: src/behavior.ts [hunk 2]
```diff
@@ -25,45 +25,63 @@
...
         this.interactivityService = options.interactivityService;
+        this.selectionHandler = selectionHandler;
 
         this.series = options.series;
 
-        this.selection.on("click", (series: StreamGraphSeries) => {
-            selectionHandler.handleSelection(
-                this.series[(<any>series).index],
-                (getEvent() as MouseEvent).ctrlKey);
+        this.selection.on("contextmenu", (datum) => {
```
Comment (MulyukovAidar) [inline src/behavior.ts:67]: private handleContextMenu = () => {
        this.svg.on('contextmenu', (event: PointerEvent, dataPoint) => {
            this.selectionManager.showContextMenu(dataPoint ? dataPoint.identity : {}, {
                x: event.clientX,
                y: event.clientY
            });
            event.preventDefault();
        });
    }

File: src/streamGraphSettingsModel.ts [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+        name: "showAxisTitle",
+        displayNameKey: "Visual_Title",
+        displayName: "Title",
+        value: true,
+        topLevelToggle: false
+    });
+}
+
+const positionOptions : IEnumMember[] = [
+    {value : LegendPosition[LegendPosition.Top], displayName : "Top"},
```
Comment (MulyukovAidar) [inline src/streamGraphSettingsModel.ts:74]: how is this connected to string resources?

File: test.webpack.config.js [hunk 4]
```diff
@@ -35,7 +35,10 @@ module.exports = {
                     {
                         loader: 'less-loader',
                         options: {
-                            paths: [path.resolve(__dirname, 'node_modules')]
+                            // paths: [path.resolve(__dirname, 'node_modules')],
```
Comment (MulyukovAidar) [inline test.webpack.config.js:38]: ?

File: test/helpers/helpers.ts [hunk 5]
```diff
@@ -37,18 +37,18 @@ export function areColorsEqual(firstColor: string, secondColor: string): boolean
 }
 
 export function isColorAppliedToElements(
-    elements: JQuery[],
-    color?: string,
+    elements,
```
Comment (MulyukovAidar) [inline test/helpers/helpers.ts:40]: add TypeScript types

File: test/visualBuilder.ts [hunk 6]
```diff
@@ -44,69 +44,55 @@ export class StreamGraphBuilder extends VisualBuilderBase<StreamGraph> {
         this.visual.update(options);
     }
 
-    public get mainElement(): JQuery {
-        return this.element.children("svg.streamGraph");
+    public get mainElement(): HTMLElement {
```
Comment (MulyukovAidar) [inline test/visualBuilder.ts:47]: this seems to be :SVGElement type

File: test/visualData.ts [hunk 7]
```diff
@@ -68,7 +68,7 @@ export class ProductSalesByDateData extends TestDataViewBuilder {
         let array: number[] = [];
         const lenght: number = valuesArray.length;
         for (let i: number = 0; i < lenght; i++) {
-            array[i] = null;
+            array[i] = null!;
```
Comment (MulyukovAidar) [inline test/visualData.ts:71]: explain this please

File: test/visualTest.ts [hunk 8]
```diff
@@ -50,48 +50,51 @@ import {
...
-                .each(function (index: number, element: Element) {
-                    let nanLocation: number = ($(element).attr("d")).indexOf("NaN");
-
-                    expect(nanLocation !== -1).toBeFalsy();
-                });
+            const dataPointsContainer = document.querySelector(".streamGraph .dataPointsContainer");
+            const paths = dataPointsContainer!.querySelectorAll("path");
+            paths.forEach(function (element, index) {
+                let nanLocation = element.getAttribute("d")!.indexOf("NaN");
+                expect(nanLocation !== -1).toBeFalsy();
```
Comment (MulyukovAidar) [inline test/visualTest.ts:91]: why not expect(nanLocation).toBe(-1)?

File: test/visualTest.ts [hunk 9]
```diff
@@ -628,7 +685,9 @@ describe("StreamGraph", () => {
 
             it("should use stroke style", (done) => {
                 visualBuilder.updateRenderTimeout(dataView, () => {
-                    const layers: JQuery<any>[] = visualBuilder.layers.toArray().map($);
+                    const layers = Array.from(visualBuilder.layers).map(element => {
```
Comment (MulyukovAidar) [inline test/visualTest.ts:688]: why do you need this .map?

File: src/behavior.ts (hunk 1 above)
Comment (snapp1) [reply src/behavior.ts:45]: yes

File: src/behavior.ts (hunk 2 above)
Comment (snapp1) [reply src/behavior.ts]: resolved

File: src/streamGraphSettingsModel.ts (hunk 3 above)
Comment (snapp1) [reply src/streamGraphSettingsModel.ts:74]: these are strings that will be used for dropdown translation

File: test.webpack.config.js (hunk 4 above)
Comment (snapp1) [reply test.webpack.config.js]: resolved

File: test/visualBuilder.ts (hunk 6 above)
Comment (snapp1) [reply test/visualBuilder.ts]: resolved

File: test/visualData.ts (hunk 7 above)
Comment (snapp1) [reply test/visualData.ts]: resolved

File: test/visualTest.ts (hunk 8 above)
Comment (snapp1) [reply test/visualTest.ts]: resolved

File: test/visualTest.ts (hunk 9 above)
Comment (snapp1) [reply test/visualTest.ts]: resolved

File: src/visual.ts [hunk 10]
```diff
@@ -699,43 +770,47 @@ export class StreamGraph implements IVisual {
 
             this.xAxisProperties = AxisHelper.createAxis(axisOptions);
 
-            this.axisX.call(this.xAxisProperties.axis);
+            this.axisX.call(<any>this.xAxisProperties.axis);
```
Comment (MulyukovAidar) [inline src/visual.ts:773]: remove any

File: src/visual.ts [hunk 11]
```diff
@@ -699,43 +770,47 @@ export class StreamGraph implements IVisual {
...
                 StreamGraph.AxisTextNodeDYForAngel0
             ];
 
-            const xAxisTextNodes: Selection<d3.BaseType, any, any, any> = this.axisX.selectAll("text");
+            const xAxisTextNodes: Selection<any, any, any, any> = this.axisX.selectAll("text");
 
+            xAxisTextNodes.call(StreamGraph.wordBreak, this.xAxisProperties, StreamGraph.XAxisLabelSize);
+            
+            /* eslint-disable-next-line prefer-spread */
             this.setTextNodesPosition.apply(this, [xAxisTextNodes].concat(transformParams));
```
Comment (MulyukovAidar) [inline src/visual.ts:791]: refactor function call

File: src/visual.ts [hunk 12]
```diff
@@ -898,66 +973,70 @@ export class StreamGraph implements IVisual {
...
-        const area: d3.Area<any> = d3.area<StreamDataPoint>()
-            .curve(d3.curveCatmullRom.alpha(0.5))
+        let areaVar: Area<any> = area<StreamDataPoint>()
             .x((d, i) => xScale(i))
             .y0(d => yScale(d[0]))
             .y1(d => yScale(d[1]))
             .defined(d => StreamGraph.isNumber(d[0]) && StreamGraph.isNumber(d[1]));
+        
+        if(this.data.formattingSettings.enableGraphCurvatureCardSettings.enabled.value) {
+            areaVar = areaVar.curve(curveCatmullRom.alpha(this.data.formattingSettings.enableGraphCurvatureCardSettings.value.value / 1.0))
```
Comment (MulyukovAidar) [inline src/visual.ts:1028]: division by 1?

**Review by MulyukovAidar — APPROVED** [review]

**Pull request merged**